pilot/pkg/networking/core/cluster_builder.go

```diff
@@ -607,6 +620,37 @@ func (cb *ClusterBuilder) setUpstreamProtocol(cluster *clusterWrapper, port *mod
 		setH2Options(cluster)
 		return
 	}
+	// Preserve HTTP/1.x traffic header case
```

Comment:
This comes after the options are set which is kind of confusing from a readability perspective. Can it be moved up?

Reply:
Not sure what you mean as this is the beginning of the logic in `setUpstreamProtocol`

Reply:
Nevermind, I see this is different now